src/services/completions.ts

```diff
@@ -113,7 +113,10 @@ namespace ts.Completions {
 
         if (keywordFilters !== KeywordCompletionFilters.None) {
             const entryNames = arrayToSet(entries, e => e.name);
-            for (const keywordEntry of getKeywordCompletions(keywordFilters)) {
+            const keywordsCompletion = isSourceFileJS(sourceFile) && keywordFilters !== KeywordCompletionFilters.TypeKeywords

```

Comment:
Can we create separate filters in all places that set keyword filters so they are handled now and later correctly. It should be done by checking isSourceFileJS and setting new keywordFilter eg.: `ClassElementKeywordsJS` etc.

Reply:
Yes, please check am I doing it right @sheetalkamat 😄